src/Libraries/Nop.Services/Plugins/DbPluginsInfo.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Data;
+using System.Data.SqlClient;
+using Newtonsoft.Json;
+using Nop.Core.Data;
+using Nop.Core.Infrastructure;
+
+namespace Nop.Services.Plugins
+{
+    public class DbPluginsInfo : PluginsInfo
```

Comment:
we do not use direct calls to the database, except for the service commands(backup, restore, reindex, etc.)

Reply:
Being that at the time this call is made we do not have a DBContext how would you suggest this is done?

Reply:
In the discussion to the task, I tried to explain to you why we do not store information about the plugins in the database, if we had a ready working idea, we would solve not only this task but also a number of related one, such as updating plugins, control version and so on. But we stell have no such solution.

Reply:
@skoshelev just so I understand - since your team does not want to write directly to the DB outside of EF the choice was made to not store plugins in the DB? 

Basically, as of 4.2, there is no method of keeping a NopCommerce web server farm in sync when it comes to installed plugins other than running a Redis server. We cannot use Redis caching as there are huge performance issues, so we must run an entire redis server just so we can keep installed plugins in sync -- and we must do it this way because we don't want to access the DB outside of EF. 

No offence, but I really don't understand this logic. Maybe I am one of the only ones running NopCommerce on more than 1 web server?

Reply:
I run three web servers in my environment, which some might say is overkill. I do however see spikes of 400+ active customers on my site at times and I don't experience downtime for every Windows update as I'm load balanced.

As for Redis I'd love to have it operating 100% so I could maximize caching between the 3 servers and better yet I don't have to bring down 2 of 3 servers to install/uninstall plugins. I'm on @toddca side regarding this issue.

Do you know if there is any immediate plans to resolve the Redis issues? If not maybe it shouldn't be acknowledged that NopCommerce supports web farms until the issues can be resolved. I certainly fell for the release notes that it was supported and purchased servers and equipment believing NopCommerce was ready for web farms.

Reply:
As I already wrote, problems with redise are solved by increasing the waiting time in the connection string, we also have plans to switch to an asynchronous call system, which should increase the performance of useing the redise db. As for your question regarding the use of direct queries, now we are not ready to implement such an approach, especially in light of our tasks to support other databases. We have a desire to fully transfer the information about plugins to the database, but not in version 4.20